core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -732,6 +732,21 @@ package object config {
       .checkValue(v => v > 0, "The max failures should be a positive value.")
       .createWithDefault(40)
 
+  private[spark] val UNSAFE_EXCEPTION_ON_MEMORY_LEAK =
+    ConfigBuilder("spark.unsafe.exceptionOnMemoryLeak")
+      .booleanConf
+      .createWithDefault(false)
+
+  private[spark] val UNSAFE_SORTER_SPILL_READ_AHEAD_ENABLED =
+    ConfigBuilder("spark.unsafe.sorter.spill.read.ahead.enabled")
+      .booleanConf
+      .createWithDefault(true)
+
+  private[spark] val UNSAFE_SORTER_SPILL_READER_BUFFER_SIZE =
+    ConfigBuilder("spark.unsafe.sorter.spill.reader.buffer.size")
+      .bytesConf(ByteUnit.BYTE)
```

Comment:
Shall we add `checkValue`? Previously, the values beyond boundary are replaced back to the default value with warnings in `UnsafeSorterSpillReader`. It would be great to check here since we have a configuration now.
```
    if (bufferSizeBytes > MAX_BUFFER_SIZE_BYTES || bufferSizeBytes < DEFAULT_BUFFER_SIZE_BYTES) {
```

Hi, @gatorsmile . If we use `checkValue` here, do we need to add a behavior change doc?
IMO, these three configuration can be `.internal()` configuration and we can skip the behavior change doc for these.

Reply:
thanks, done

Reply:
Thank you for updating. The followings remain. Could you add `.internal()`?
> IMO, these three configuration can be `.internal()` configuration and we can skip the behavior change doc for these.